sample/authentication/crypto.go

```diff
@@ -163,10 +185,45 @@ func (au *SGXUSIGAuthenticationScheme) VerifyAuthenticationTag(m []byte, sig []b
 		return fmt.Errorf("failed to unmarshal UI: %v", err)
 	}
 
-	usigID, err := sgxusig.MakeID(ui.Epoch, pubKey)
+	fingerprint, err := makeUSIGKeyFingerprint(pubKey)
+	if err != nil {
+		return fmt.Errorf("Failed to calculate USIG key fingerprint: %s", err)
+	}
+
+	au.lock.Lock()
+	defer au.lock.Unlock()
+
+	// Capture the epoch value received with the first valid UI to
+	// dynamically determine the full USIG identity. Note that
+	// this relies on the assumption that all peer replicas are
+	// initially correct, each use its unique USIG key pairs per
+	// consensus protocol instance, and generate the first UI
+	// using a single USIG instance per replica. Moreover, those
+	// first UIs are assumed to be received and processed by
+	// correct replicas before any replica becomes faulty in a
+	// sense that it starts generating and sending UIs using
+	// another USIG instance initialized with the same sealed key
+	// pair.
+	//
+	// Those assumptions might be too strong in some environments.
+	// In that case, there should be a way for all correct
+	// replicas to agree on a single USIG instance identity per
```

Comment:
If there is a correct bootstrap/initialization stage, replicas are not required to be all correct. 
Therefore, "in that case, a correct bootstrap is required for all replicas to first agree on...".

Reply:
Exactly, there would be no need for all replicas to be initially correct. That is why only the correct replicas need to agree; up to f replicas could be faulty from the beginning and do not agree. I didn't want to emphasize how it should be achieved. It could be achieved at bootstrap phase or maybe some other way. But I'm considering to change the wording to: "In that case, all correct replicas are required to agree ..." Would that make sense?